src/Engine/StateTrackers/DSSonjaDebuffTracker.java

```diff
@@ -0,0 +1,66 @@
+package Engine.StateTrackers;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import Engine.Army;
+import Engine.GameInstance;
+import Engine.Combat.CombatContext;
+import Engine.GameEvents.GameEventQueue;
+import Engine.UnitMods.UnitModifierWithDefaults;
+import Terrain.MapMaster;
+import Units.UnitContext;
+import lombok.var;
+
+public class DSSonjaDebuffTracker extends StateTracker implements UnitModifierWithDefaults
+{
+  private static final long serialVersionUID = 1L;
+
+  // Stores the debuff strength to be applied to each army this turn; recalculated on turn change and Sonja (S)COP, since cart Sonja's debuff lasts to the end of the turn she's routed [original research].
+  public HashMap<Army, Integer> debuffMap = new HashMap<>();
+  public HashSet<CommandingOfficers.AW3.YC.Sonja> debuffers = new HashSet<>();
```

Comment:
AW3 Sonja is the only one who applies the effect to everyone?

Reply:
Yeah. I consider it a bug, so it's a cart-CO-only feature.